From 946381fd6fdabedf2b5a1d8647a49e3da8f1894e Mon Sep 17 00:00:00 2001
From: Baruch Siach <baruch@tkos.co.il>
Date: Fri, 23 Jun 2023 11:28:02 +0200
Subject: [PATCH 1/8] nss-dp: Drop _nocache variants of ioremap()

ioremap_nocache was made equivelant to ioremap and was dropped.

Signed-off-by: Baruch Siach <baruch@tkos.co.il>
Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 hal/dp_ops/edma_dp/edma_v1/edma_data_plane.c | 2 +-
 hal/gmac_ops/syn/gmac/syn_if.c               | 2 +-
 hal/soc_ops/ipq50xx/nss_ipq50xx.c            | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

--- a/hal/dp_ops/edma_dp/edma_v1/edma_data_plane.c
+++ b/hal/dp_ops/edma_dp/edma_v1/edma_data_plane.c
@@ -279,7 +279,7 @@ int edma_init(void)
 	/*
 	 * Remap register resource
 	 */
-	edma_hw.reg_base = ioremap_nocache((edma_hw.reg_resource)->start,
+	edma_hw.reg_base = ioremap((edma_hw.reg_resource)->start,
 				resource_size(edma_hw.reg_resource));
 	if (!edma_hw.reg_base) {
 		pr_warn("Unable to remap EDMA register memory.\n");
--- a/hal/gmac_ops/syn/gmac/syn_if.c
+++ b/hal/gmac_ops/syn/gmac/syn_if.c
@@ -806,7 +806,7 @@ static void *syn_init(struct nss_gmac_ha
 	 * Populate the mac base addresses
 	 */
 	shd->nghd.mac_base =
-		devm_ioremap_nocache(&dp_priv->pdev->dev, res->start,
+		devm_ioremap(&dp_priv->pdev->dev, res->start,
 							resource_size(res));
 	if (!shd->nghd.mac_base) {
 		netdev_dbg(ndev, "ioremap fail.\n");
--- a/hal/soc_ops/ipq50xx/nss_ipq50xx.c
+++ b/hal/soc_ops/ipq50xx/nss_ipq50xx.c
@@ -89,7 +89,7 @@ static void nss_dp_hal_tcsr_set(void)
 			pr_err("%s: SCM TCSR write error: %d\n", __func__, err);
 		}
 	} else {
-		tcsr_addr = ioremap_nocache((tcsr_base + TCSR_GMAC_AXI_CACHE_OVERRIDE_OFFSET),
+		tcsr_addr = ioremap((tcsr_base + TCSR_GMAC_AXI_CACHE_OVERRIDE_OFFSET),
 						TCSR_GMAC_AXI_CACHE_OVERRIDE_REG_SIZE);
 		if (!tcsr_addr) {
 			pr_err("%s: ioremap failed\n", __func__);
